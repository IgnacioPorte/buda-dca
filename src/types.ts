type HistoricalPrice = {
  price: number;
  date: string;
};

type MonthlyDCAInvestment = {
  price: number;
  date: string;
  fiatInvestment: number;
  cryptoPortfolio: number;
  portfolioFiatValue: number;
  gain: number;
  gainPercentage: number;
};


export type { HistoricalPrice, MonthlyDCAInvestment };