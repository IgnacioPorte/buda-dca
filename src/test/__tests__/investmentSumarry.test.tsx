import InvestmentSummary from "@/components/investmentSummary";
import { MonthlyDCAInvestment } from "@/types";
import { describe, expect, test } from "vitest";
import { render } from "@testing-library/react";

const mockData: MonthlyDCAInvestment[] = [
  {
    cryptoPortfolio: 0.1,
    date: "2021-01",
    fiatInvestment: 100,
    gain: 0,
    gainPercentage: 0,
    portfolioFiatValue: 150,
    price: 150,
  },
  {
    cryptoPortfolio: 0.2,
    date: "2021-02",
    fiatInvestment: 200,
    gain: 50,
    gainPercentage: 25,
    portfolioFiatValue: 250,
    price: 250,
  },
];

describe("LinearChart Component Tests", () => {
  test("renders without crashing", () => {
    expect(render(<InvestmentSummary data={[]} />)).toMatchSnapshot();
  });

  test("renders chart with data", () => {
    const investmentSummary = render(<InvestmentSummary data={mockData} />);
    expect(investmentSummary).toMatchSnapshot();
    expect(investmentSummary.getByText("25.00%")).toBeDefined();
    expect(investmentSummary.getByText("$ 200.00")).toBeDefined();
    expect(investmentSummary.getByText("$ 250.00")).toBeDefined();
  });
});
