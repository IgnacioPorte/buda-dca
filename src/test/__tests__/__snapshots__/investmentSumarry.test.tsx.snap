// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LinearChart Component Tests > renders chart with data 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-row justify-center gap-5 my-2 flex-wrap"
      >
        <div
          class="flex flex-col justify-center border-2 border-gray-1 my-2 p-2 w-3/4 sm:w-1/4"
        >
          <p
            class="text-2xl"
          >
            Dinero Invertido
          </p>
          <p
            class="text-lg"
          >
            200.00
          </p>
        </div>
        <div
          class="border-2 border-gray-1 my-2 p-1 w-3/4 sm:w-1/4"
        >
          <svg
            class="w-10 h-10 mx-auto"
            fill="none"
            height="15"
            viewBox="0 0 15 15"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M2.14645 11.1464C1.95118 11.3417 1.95118 11.6583 2.14645 11.8536C2.34171 12.0488 2.65829 12.0488 2.85355 11.8536L6.85355 7.85355C7.04882 7.65829 7.04882 7.34171 6.85355 7.14645L2.85355 3.14645C2.65829 2.95118 2.34171 2.95118 2.14645 3.14645C1.95118 3.34171 1.95118 3.65829 2.14645 3.85355L5.79289 7.5L2.14645 11.1464ZM8.14645 11.1464C7.95118 11.3417 7.95118 11.6583 8.14645 11.8536C8.34171 12.0488 8.65829 12.0488 8.85355 11.8536L12.8536 7.85355C13.0488 7.65829 13.0488 7.34171 12.8536 7.14645L8.85355 3.14645C8.65829 2.95118 8.34171 2.95118 8.14645 3.14645C7.95118 3.34171 7.95118 3.65829 8.14645 3.85355L11.7929 7.5L8.14645 11.1464Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
          <p
            class="text-sm text-center text-green-500"
          >
            +
            50.00
          </p>
           
          <p
            class="text-sm text-center  text-green-500"
          >
            25.00
            %
          </p>
        </div>
        <div
          class="flex flex-col justify-center border-2 border-gray-1 my-2 p-1 w-3/4 sm:w-1/4"
        >
          <p
            class="text-2xl"
          >
            Valor del portafolio
          </p>
          <p
            class="text-lg"
          >
            250.00
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-row justify-center gap-5 my-2 flex-wrap"
    >
      <div
        class="flex flex-col justify-center border-2 border-gray-1 my-2 p-2 w-3/4 sm:w-1/4"
      >
        <p
          class="text-2xl"
        >
          Dinero Invertido
        </p>
        <p
          class="text-lg"
        >
          200.00
        </p>
      </div>
      <div
        class="border-2 border-gray-1 my-2 p-1 w-3/4 sm:w-1/4"
      >
        <svg
          class="w-10 h-10 mx-auto"
          fill="none"
          height="15"
          viewBox="0 0 15 15"
          width="15"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M2.14645 11.1464C1.95118 11.3417 1.95118 11.6583 2.14645 11.8536C2.34171 12.0488 2.65829 12.0488 2.85355 11.8536L6.85355 7.85355C7.04882 7.65829 7.04882 7.34171 6.85355 7.14645L2.85355 3.14645C2.65829 2.95118 2.34171 2.95118 2.14645 3.14645C1.95118 3.34171 1.95118 3.65829 2.14645 3.85355L5.79289 7.5L2.14645 11.1464ZM8.14645 11.1464C7.95118 11.3417 7.95118 11.6583 8.14645 11.8536C8.34171 12.0488 8.65829 12.0488 8.85355 11.8536L12.8536 7.85355C13.0488 7.65829 13.0488 7.34171 12.8536 7.14645L8.85355 3.14645C8.65829 2.95118 8.34171 2.95118 8.14645 3.14645C7.95118 3.34171 7.95118 3.65829 8.14645 3.85355L11.7929 7.5L8.14645 11.1464Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
        <p
          class="text-sm text-center text-green-500"
        >
          +
          50.00
        </p>
         
        <p
          class="text-sm text-center  text-green-500"
        >
          25.00
          %
        </p>
      </div>
      <div
        class="flex flex-col justify-center border-2 border-gray-1 my-2 p-1 w-3/4 sm:w-1/4"
      >
        <p
          class="text-2xl"
        >
          Valor del portafolio
        </p>
        <p
          class="text-lg"
        >
          250.00
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`LinearChart Component Tests > renders without crashing 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-row justify-center gap-5 my-2 flex-wrap"
      >
        <div
          class="animate-pulse bg-primary/10 h-24 w-3/4 sm:w-1/4 rounded-xl"
        />
        <div
          class="animate-pulse bg-primary/10 h-24 w-3/4 sm:w-1/4 rounded-xl"
        />
        <div
          class="animate-pulse bg-primary/10 h-24 w-3/4 sm:w-1/4 rounded-xl"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-row justify-center gap-5 my-2 flex-wrap"
    >
      <div
        class="animate-pulse bg-primary/10 h-24 w-3/4 sm:w-1/4 rounded-xl"
      />
      <div
        class="animate-pulse bg-primary/10 h-24 w-3/4 sm:w-1/4 rounded-xl"
      />
      <div
        class="animate-pulse bg-primary/10 h-24 w-3/4 sm:w-1/4 rounded-xl"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
